src/bitcoind.cpp

```diff
@@ -143,6 +135,14 @@ bool AppInit(int argc, char* argv[])
             return false;
         }
 
+        // Handle setting of allowed-deprecated features as early as possible
+        // so that it's possible for other initialization steps to respect them.
+        auto deprecationError = LoadAllowedDeprecatedFeatures();
+        if (deprecationError.has_value()) {
+            fprintf(stderr, "%s", deprecationError.value().c_str());
```

Comment:
Preexisting bug/non-portable code: stderr is line-buffered, so the message is not guaranteed to be flushed (it is implementation-dependent whether it is flushed on abort).
```suggestion
            fprintf(stderr, "%s\n", deprecationError.value().c_str());
```